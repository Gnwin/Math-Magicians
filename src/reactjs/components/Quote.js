import React from 'react';

const Quote = () => (
  <div className="quote" data-testid="quote-1">
    <p className="quote-msg" data-testid="quote-2">Mathematics is not about numbers, equations, computations or algorithms: it is about understanding. -William Paul Thurston</p>
  </div>
);

export default Quote;
