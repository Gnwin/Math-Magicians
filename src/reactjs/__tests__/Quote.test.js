import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import Quote from '../components/Quote';

describe('it tests the Quote Component', () => {
  it('renders quote component correctly', () => {
    const tree = renderer
      .create(<Quote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('checks if quotemsg is a descendant of quoteComponent', () => {
    render(<Quote />);
    const quoteComponent = screen.getByTestId('quote-1');
    const quotemsg = screen.getByTestId('quote-2');
    expect(quoteComponent).toContainElement(quotemsg);
  });

  it('checks if quotemsg has a text content', () => {
    render(<Quote />);
    const quotemsg = screen.getByTestId('quote-2');
    expect(quotemsg).toHaveTextContent('Mathematics is not about numbers, equations, computations or algorithms: it is about understanding. -William Paul Thurston');
  });
});
