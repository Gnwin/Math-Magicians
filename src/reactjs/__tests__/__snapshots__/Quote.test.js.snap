// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the Quote Component renders quote component correctly 1`] = `
<div
  className="quote"
  data-testid="quote-1"
>
  <p
    className="quote-msg"
    data-testid="quote-2"
  >
    Mathematics is not about numbers, equations, computations or algorithms: it is about understanding. -William Paul Thurston
  </p>
</div>
`;
